### PR TITLE
Implement basic find by prefix feature

### LINKED ISSUE #69: Implement Finding Feature
As a user, I want to find potential donors for a patient so I can help the patient find a donor as fast as possible.

@TheRealRavager Previously, I think this feature was assigned to you. See if you want to track your progress of this feature with this issue or if you want to make further changes.

### PR DESCRIPTION
**Pull request by TheRealRavager**
Allows users to find persons by passing in prefixes such as: `n/, ic/, p/, a/, t/, pr/, b/, d/, tt/, exp/, o/`

Fixes #69

### CODE REVIEW

**Comment by TheRealRavager** [comment]
Current Bugs:
- [ ] Find displays people who do not possess that prefix at all (e.g. `bloodType` for `Doctor`). Possibly caused by early check of blood_type being empty. Proposed solution: For type specific, should return false if not of that type.
- [ ] Find by tissue type requires **exact** matching of tissue types (including order)
- [ ] Find by `organExpiryDate` not working.
- [ ] Find by `organ` does not return donors
- [ ] Wrong **preamble** format causes find to return all people instead of giving user feedback

**Comment by C-likethis123** [comment]
> Current Bugs:
> 
> * [x]  Find displays people who do not possess that prefix at all (e.g. `bloodType` for `Doctor`). Possibly caused by early check of blood_type being empty. Proposed solution: For type specific, should return false if not of that type.
> * [x]  ~Find by `priority` not working.~
> * [ ]  Find by tissue type requires **exact** matching of tissue types (including order) (Feature or bug?)
> * [x]  ~Find by `organExpiryDate` not working. Need to check toString method.~
> * [x]  Find by `organ` does not return donors
> * [x]  Wrong **preamble** format causes find to return all people instead of giving user feedback

**Comment by TheRealRavager** [comment]
> > Current Bugs:
> > 
> > * [x]   Find displays people who do not possess that prefix at all (e.g. `bloodType` for `Doctor`). Possibly caused by early check of blood_type being empty. Proposed solution: For type specific, should return false if not of that type.
> > * [x]   ~Find by `priority` not working.~
> > * [ ]   Find by tissue type requires **exact** matching of tissue types (including order) (Feature or bug?)
> > * [x]   ~Find by `organExpiryDate` not working. Need to check toString method.~
> > * [x]   Find by `organ` does not return donors
> > * [x]   Wrong **preamble** format causes find to return all people instead of giving user feedback
> 
> With regards to finding by tissue type, I propose that the order of the tissue types should not matter. The search could pull out people who do not have the exact same tissue types, but the closer the search the higher up it turns up in the search results.

Alright, have discussed with @WilliamRyank too, find by tissue type will display all those persons who have all the input tissue types.

File: src/main/java/organice/logic/commands/FindCommand.java [hunk 1]
```diff
@@ -4,24 +4,25 @@
...
+            + "List of Prefixes: n/, ic/, p/, a/, t/, pr/, b/, d/, tt/, exp/, o/"
+            + "Parameters: PREFIX/KEYWORD [MORE_PREFIX-KEYWORD_PAIRS]...\n"
+            + "Example: " + COMMAND_WORD + " n/alice t/doctor";
 
-    private final NameContainsKeywordsPredicate predicate;
+    private final PersonContainsPrefixesPredicate predicate;
 
-    public FindCommand(NameContainsKeywordsPredicate predicate) {
+    public FindCommand(PersonContainsPrefixesPredicate predicate) {
         this.predicate = predicate;
```
Comment (WilliamRyank) [inline src/main/java/organice/logic/commands/FindCommand.java]: allowed null here?

File: src/test/java/organice/logic/commands/CommandTestUtil.java [hunk 2]
```diff
@@ -180,6 +181,8 @@ public static void assertCommandSuccess(Command command, Model actualModel, Comm
             Model expectedModel) {
         try {
             CommandResult result = command.execute(actualModel);
+            System.out.println(expectedCommandResult.getFeedbackToUser());
+            System.out.println(result.getFeedbackToUser());
```
Comment (WilliamRyank) [inline src/test/java/organice/logic/commands/CommandTestUtil.java]: Forgot to remove i think

File: src/main/java/organice/logic/commands/FindCommand.java (hunk 1 above)
Comment (TheRealRavager) [reply src/main/java/organice/logic/commands/FindCommand.java]: Changed

File: src/test/java/organice/logic/commands/CommandTestUtil.java (hunk 2 above)
Comment (TheRealRavager) [reply src/test/java/organice/logic/commands/CommandTestUtil.java]: Removed. Thanks!

**Comment by TheRealRavager** [comment]
I just found a bug, `find n/Alice Carl Elle t/doctor patient` fails to display Alice (who is a doctor) and displays Elle (a donor)

File: src/test/java/organice/logic/parser/AddressBookParserTest.java [hunk 3]
```diff
@@ -70,10 +70,12 @@ public void parseCommand_exit() throws Exception {
 
     @Test
     public void parseCommand_find() throws Exception {
-        List<String> keywords = Arrays.asList("foo", "bar", "baz");
+        String searchParams = "find n/Alice ic/S1111111A t/doctor";
```
Comment (WilliamRyank) [inline src/test/java/organice/logic/parser/AddressBookParserTest.java]: Should use constant but its ok for now like `commandText = FindCommand.Command_Word .....`

Comment (TheRealRavager) [reply src/test/java/organice/logic/parser/AddressBookParserTest.java]: changed

**Pull request merged**